audiomate/corpus/io/lingualibre.py

```diff
@@ -0,0 +1,164 @@
+    "que": "https://lingualibre.org/datasets/Q388-que-Quechua.zip",
+    "rus": "https://lingualibre.org/datasets/Q129-rus-Russian.zip",
+    "sat": "https://lingualibre.org/datasets/Q339-sat-Santali.zip",
+    "shy": "https://lingualibre.org/datasets/Q4901-shy-Shawiyalanguage.zip",
+    "spa": "https://lingualibre.org/datasets/Q386-spa-Spanish.zip",
+    "srr": "https://lingualibre.org/datasets/Q101-srr-Serer.zip",
+    "swe": "https://lingualibre.org/datasets/Q44-swe-Swedish.zip",
+    "tam": "https://lingualibre.org/datasets/Q127-tam-Tamil.zip",
+    "tay": "https://lingualibre.org/datasets/Q51302-tay-Atayal.zip",
+    "tel": "https://lingualibre.org/datasets/Q39-tel-Telugu.zip",
+    "tgl": "https://lingualibre.org/datasets/Q169-tgl-Tagalog.zip",
+    "vie": "https://lingualibre.org/datasets/Q208-vie-Vietnamese.zip",
+    "zho": "https://lingualibre.org/datasets/Q130-zho-Chinese.zip",
+}
+
+
+# ==================================================================================================
+
+
+class LinguaLibreDownloader(base.CorpusDownloader):
+    def __init__(self, lang="deu"):
+        if lang in urls.keys():
+            self.url = urls[lang]
+        else:
+            msg = "There is no lingualibre URL present for language {}!"
+            raise ValueError(msg.format(lang))
+
+    @classmethod
+    def type(cls):
+        return "lingualibre"
+
+    # ==============================================================================================
+
+    def _download(self, target_path):
+        """ Download the data to target_path """
+
+        os.makedirs(target_path, exist_ok=True)
+        tmp_file = os.path.join(target_path, "tmp_lingualibre.zip")
+
+        download.download_file(self.url, tmp_file, num_threads=1)
+        download.extract_zip(tmp_file, target_path)
+        os.remove(tmp_file)
+
+
+# ==================================================================================================
+
+
+class LinguaLibreReader(base.CorpusReader):
+    """ Reader for collections of lingualibre audio data.
+    The reader expects extracted .zip files in the given folder """
+
+    @classmethod
+    def type(cls):
+        return "lingualibre"
+
+    def _check_for_missing_files(self, path):
+        return []
+
+    # ==============================================================================================
+
+    def _load(self, path):
+        corpus = audiomate.Corpus(path=path)
+
+        all_files = []
+        for dirpath, dirs, files in os.walk(path):
+            for file in files:
+                all_files.append([dirpath, file])
+
+        for dirpath, file in logger.progress(all_files):
+            file_path = os.path.join(dirpath, file)
+            issuer_idx = os.path.basename(dirpath)
+            transcription = os.path.splitext(os.path.basename(file_path))[0]
+
+            # Create files ...
+            file_idx = hashlib.sha256(file_path.encode("utf-8")).hexdigest()
```

Comment:
Why do use the hash? Are there a lot of special characters in the path/filename?

Reply:
Yes the filename contains the transcription and the path has also many special characters like spaces or commas.